common/src/ui/BoxSelectionTool.cpp

```
@@ -0,0 +1,302 @@
+        
+        bool nodeSelected = false;
+        
+        // Determine if the node is within the selection range based on the current view projection
+        if (majorAxis == vm::axis::z) {
+          // Top view - projected onto the XY plane
+          nodeSelected = nodeCenter.x() >= m_selectionBounds.min.x() &&
+                        nodeCenter.x() <= m_selectionBounds.max.x() &&
+                        nodeCenter.y() >= m_selectionBounds.min.y() &&
+                        nodeCenter.y() <= m_selectionBounds.max.y();
+        } else if (majorAxis == vm::axis::y) {
+          // Front view - projected onto the XZ plane
+          nodeSelected = nodeCenter.x() >= m_selectionBounds.min.x() &&
+                        nodeCenter.x() <= m_selectionBounds.max.x() &&
+                        nodeCenter.z() >= m_selectionBounds.min.z() &&
+                        nodeCenter.z() <= m_selectionBounds.max.z();
+        } else {
+          // Side view - projected onto the YZ plane
+          nodeSelected = nodeCenter.y() >= m_selectionBounds.min.y() &&
+                        nodeCenter.y() <= m_selectionBounds.max.y() &&
+                        nodeCenter.z() >= m_selectionBounds.min.z() &&
+                        nodeCenter.z() <= m_selectionBounds.max.z();
+        }
+        
+        if (nodeSelected) {
+          selectedNodes.push_back(node);
+        }
+      }
+      
+      // Filter out selectable nodes
+      auto selectableNodes = mdl::collectSelectableNodes(
+        selectedNodes, document->editorContext());
+        
+      // If there are selectable nodes, create a selection transaction
+      if (!selectableNodes.empty()) {
+        auto transaction = Transaction{*document, "Box Select"};
+        document->deselectAll();
+        document->selectNodes(selectableNodes);
+        transaction.commit();
+      }
+      
+      // Clear the renderer
+      m_renderer.clear();
+    } catch (const std::exception&) {
```

I'm not sure whether anything you are calling in the `try` block can throw - but if it does, is it correct to just ignore the exceptions?